src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs

```diff
@@ -2944,103 +2783,9 @@ private static int LengthInBufferCellsFE(char c, ref HWND hwnd, ref HDC hDC, ref
-                        int err = Marshal.GetLastWin32Error();
-                        //Don't throw exception so that output can continue
-                        tracer.TraceError("Win32 Error 0x{0:X} occurred when getting the window handle to the console.",
-                            err);
-                        return 1;
-                    }
-                    hDC = NativeMethods.GetDC(hwnd);
-                    if ((IntPtr)0 == hDC)
-                    {
-                        int err = Marshal.GetLastWin32Error();
-                        //Don't throw exception so that output can continue
-                        tracer.TraceError("Win32 Error 0x{0:X} occurred when getting the Device Context of the console window.",
-                            err);
-                        return 1;
-                    }
-                }
-                bool result = true;
-                if (!istmInitialized)
-                {
-                    result = NativeMethods.GetTextMetrics(hDC, out tm);
-                    if (!result)
-                    {
-                        int err = Marshal.GetLastWin32Error();
-                        //Don't throw exception so that output can continue
-                        tracer.TraceError("Win32 Error 0x{0:X} occurred when getting the Text Metric of the console window's Device Context.",
-                            err);
-                        return 1;
-                    }
-                    istmInitialized = true;
-                }
-                int width;
-                result = NativeMethods.GetCharWidth32(hDC, (uint)c, (uint)c, out width);
-                if (!result)
-                {
-                    int err = Marshal.GetLastWin32Error();
-                    //Don't throw exception so that output can continue
-                    tracer.TraceError("Win32 Error 0x{0:X} occurred when getting the width of a char.",
-                        err);
-                    return 1;
-                }
-                if (width >= tm.tmMaxCharWidth)
-                {
-                    return 2;
-                }
-#endif
-            }
             tracer.WriteLine("failed to locate char {0}, return 1", (int)c);
             return 1;
         }
-
-        internal static int LengthInBufferCells(char c)
-        {
-            uint codePage = NativeMethods.GetConsoleOutputCP();
-            return LengthInBufferCells(c, codePage);
-        }
-
-        /// <summary>
-        /// From IsConsoleFullWidth in \windows\core\ntcon\server\dbcs.c
-        /// </summary>
-        /// <param name="c"></param>
-        /// <param name="codePage"></param>
-        /// <returns></returns>
-        [SuppressMessage("Microsoft.Usage", "CA1806:DoNotIgnoreMethodResults",
-            MessageId = "Microsoft.PowerShell.ConsoleControl+NativeMethods.ReleaseDC(System.IntPtr,System.IntPtr)")]
-        private static int LengthInBufferCells(char c, uint codePage)
-        {
```

Comment:
I think we still need LengthInBufferCells - but if you want to make a simplifying assumption when you can't know the font, you can use this [code](https://github.com/lzybkr/PSReadLine/blob/8edcf64663d7b7de3d9ef23435473e6529954b1e/PSReadLine/Render.cs#L512) from PSReadline.

Reply:
I'll just use the code from PSReadline.  I ended up removing all this code as the fundamental assumption in the code is that we are using a specific codePage so all the calculations ended up returning a width of 1 anyways as all the other supporting code only applied to Windows PowerShell.